packages/medusa-plugin-auth/src/auth-strategies/keycloak/store.ts

```diff
@@ -0,0 +1,81 @@
+import { Router } from 'express';
+import { ConfigModule, MedusaContainer } from '@medusajs/medusa/dist/types/global';
+import { Strategy as KeycloakStrategy } from "passport-oauth2";
+import { KeycloakOptions, Profile, ExtraParams, KEYCLOAK_STORE_STRATEGY_NAME } from './types';
+import { PassportStrategy } from '../../core/passport/Strategy';
+import { validateStoreCallback } from '../../core/validate-callback';
+import { passportAuthRoutesBuilder } from '../../core/passport/utils/auth-routes-builder';
+
+export class KeycloakStoreStrategy extends PassportStrategy(KeycloakStrategy, KEYCLOAK_STORE_STRATEGY_NAME) {
+	constructor(
+		protected readonly container: MedusaContainer,
+		protected readonly configModule: ConfigModule,
+		protected readonly strategyOptions: KeycloakOptions
+	) {
+		super({
+      authorizationURL: strategyOptions.authorizationURL,
+      tokenURL: strategyOptions.tokenURL,
+      clientID: strategyOptions.clientID,
+      clientSecret: strategyOptions.clientSecret,
+      callbackURL: strategyOptions.admin.callbackUrl,
```

Comment:
it should be strategyOptions.store.callbackUrl

Reply:
you need to add index.ts file to load plugin
`import { AuthOptions, StrategyExport } from '../../types';
import { Router } from 'express';
import { getKeycloakAdminAuthRouter, KeycloakAdminStrategy } from './admin';
import { ConfigModule, MedusaContainer } from '@medusajs/medusa/dist/types/global';
import { getKeycloakStoreAuthRouter, KeycloakStoreStrategy } from './store';

export * from './types';
export * from './admin';
export * from './store';

export default {
	load: (container: MedusaContainer, configModule: ConfigModule, options: AuthOptions): void => {
		if (options.keycloak?.admin) {
			new KeycloakAdminStrategy(container, configModule, options.keycloak);
		}

		if (options.keycloak?.store) {
			new KeycloakStoreStrategy(container, configModule, options.keycloak);
		}
	},
	getRouter: (configModule: ConfigModule, options: AuthOptions): Router[] => {
		const routers = [];

		if (options.keycloak?.admin) {
			routers.push(getKeycloakAdminAuthRouter(options.keycloak, configModule));
		}

		if (options.keycloak?.store) {
			routers.push(getKeycloakStoreAuthRouter(options.keycloak, configModule));
		}

		return routers;
	},
} as StrategyExport;
`